cognite/neat/_graph/transformers/_classic_cdf.py

```diff
@@ -305,45 +327,70 @@ def transform(self, graph: Graph) -> None:
                     instance_count = int(instance_count_res[0])  # type: ignore[index, arg-type]
                     if instance_count < self._min_relationship_types:
                         continue
-                    query = self._instances.format(
-                        namespace=self._namespace, source_type=source_type, target_type=target_type
-                    )
-                    total = instance_count if self._limit_per_type is None else self._limit_per_type
-                    for no, result in enumerate(
-                        iterate_progress_bar(graph.query(query), total=total, description="Relationships to edges")
-                    ):
-                        if self._limit_per_type is not None and no >= self._limit_per_type:
-                            break
-                        instance_id = cast(URIRef, result[0])  # type: ignore[index, misc]
-                        self._convert_relationship_to_schema(graph, instance_id, source_type, target_type)
-
-    def _convert_relationship_to_schema(
-        self, graph: Graph, instance_id: URIRef, source_type: str, target_type: str
-    ) -> None:
-        relationship_triples = cast(list[Triple], list(graph.query(f"DESCRIBE <{instance_id}>")))
+                    edge_triples = self._edge_triples(graph, source_type, target_type, instance_count, entity_lookup)
+                    add_triples_in_batch(graph, edge_triples)
+
+    def _edge_triples(
+        self,
+        graph: Graph,
+        source_type: str,
+        target_type: str,
+        instance_count: int,
+        entity_lookup: Callable[[str, str], URIRef],
+    ) -> Iterable[Triple]:
+        query = self._instances.format(namespace=self._namespace, source_type=source_type, target_type=target_type)
+        total = instance_count if self._limit_per_type is None else self._limit_per_type
```

Comment:
naming is confusing here, what total means?

Reply:
and why is it falling back to self._limit_per_type ?
Is self._limit_per_type set somewhere?